microsoft-edge/devtools-guide-chromium/memory-problems/dom-leaks.md

```diff
@@ -0,0 +1,120 @@
+
+1. Open the [demo application](https://microsoftedge.github.io/Demos/detached-elements/).
+
+1. [Open DevTools](/devtools-guide-chromium/#opening-devtools).
+
+1. Click the **+** button in the toolbar.
+
+1. Click **Detached Elements**.
+
+:::image type="complex" source="./images/open-detached-elements.msft.png" alt-text="Open the Detached Elements tool." lightbox="./images/open-detached-elements.msft.png":::
+   Open the Detached Elements tool
+:::image-end:::
+
+## Get Detached Elements
+
+The **Get Detached Elements** button in the **Detached Elements** tool will find and display all of the detached elements on the page. To find detached elements in the demo application, complete the following steps:
+
+1. Make sure you are in **Room 1** of the demo application.
+
+1. Click the **Simulate high traffic** button.
+
+1. Click **Stop** after generating some messages in the demo application.
+
+1. Click **Room 2**.
+
+1. In the **Detached Elements** tool, click **Get Detached Elements**.
+
+:::image type="complex" source="./images/get-detached-elements.msft.png" alt-text="Get Detached Elements using the Detached Elements tool." lightbox="./images/get-detached-elements.msft.png":::
+   Get Detached Elements using the Detached Elements tool.
+:::image-end:::
+
+The **Detached Elements** tool displays all of the detached elements on the page. All of the messages that were generated in **Room 1** are no longer attached to the DOM when you switch to **Room 2** in the demo application, but they are still referenced by JavaScript.  
+
+## Trigger Garbage Collection
+
+You can trigger garbage collection (GC) in the browser from the **Detached Elements** tool. Complete the following steps;
+
+1. In the **Detached Elements** tool, click **Collect Garbage**.
+
+1. Click **Get Detached Elements**.
+
+When you select **Collect Garbage**, the browser run GC. When you select **Get Detached Elements**, the **Detached Elements** tool displays all detached elements that cannot be garbage collected. These detached elements may be memory leaks if they are not going to be reused by the application.
+
+## Analyze
+
+Once you have found a detached element that the browser cannot GC, you can use the **Analyze** button in the **Detached Elements** tool to identify the JavaScript code running on the page that is still referencing the detached element. The **Analyze** button will take a heap snapshot and populate the **ID** of the detached element with its location in the heap. For more information about heap snapshots, read [Record heap snapshots](/devtools-guide-chromium/memory-problems/heap-snapshots).
+
+:::image type="complex" source="./images/analyze-detached-elements.msft.png" alt-text="Analyze Detached Elements in the Detached Elements tool." lightbox="./images/analyze-detached-elements.msft.png":::
+   Analyze Detached Elements in the Detached Elements tool
+:::image-end:::
+
+Complete the following steps:
```

Comment:
This sounds a little bit robotic. Consider: `To identify the JavaScript code that references a detached element:`.